addons/resource/tests/common.py

```diff
@@ -3,23 +3,21 @@
 
 from datetime import datetime
 
-from openerp.tests import common
+from odoo.tests import common
 
 
 class TestResourceCommon(common.TransactionCase):
 
     def setUp(self):
         super(TestResourceCommon, self).setUp()
-        cr, uid = self.cr, self.uid
-        self.context = context = dict(tz='UTC')
-
-        self.registry('res.users').write(cr, uid, uid, {'tz': 'UTC'}, context=context)
+        self.env.context = context = dict(tz='UTC')
+        self.env.user.tz = 'UTC'
 
         # Usefull models
-        self.resource_resource = self.registry('resource.resource')
-        self.resource_calendar = self.registry('resource.calendar')
-        self.resource_attendance = self.registry('resource.calendar.attendance')
-        self.resource_leaves = self.registry('resource.calendar.leaves')
+        self.Resource = self.env['resource.resource']
+        self.ResourceCalendar = self.env['resource.calendar']
+        self.ResourceAttendance = self.env['resource.calendar.attendance']
+        self.ResourceLeaves = self.env['resource.calendar.leaves']
 
         # Some demo data
         self.date1 = datetime.strptime('2013-02-12 09:08:07', '%Y-%m-%d %H:%M:%S')  # weekday() returns 1, isoweekday() returns 2
```

Comment:
How about using fields.Datetime.from_string here ? Can you check the whole file?